packages/php-db-import-export/src/Backend/Snowflake/ToFinalTable/SqlBuilder.php

```diff
@@ -184,44 +186,67 @@ public function getInsertAllIntoTargetTableCommand(
 
         $columnsSetSql = [];
 
-        /** @var SnowflakeColumn $columnDefinition */
-        foreach ($sourceTableDefinition->getColumnsDefinitions() as $columnDefinition) {
+        /** @var SnowflakeColumn $sourceColumn */
+        foreach ($sourceTableDefinition->getColumnsDefinitions() as $sourceColumn) {
             // output mapping same tables are required do not convert nulls to empty strings
             if (!$importOptions->isNullManipulationEnabled()) {
-                $columnsSetSql[] = SnowflakeQuote::quoteSingleIdentifier($columnDefinition->getColumnName());
+                $destinationColumn = $columnMap->getDestination($sourceColumn);
+                $type = $destinationColumn->getColumnDefinition()->getType();
+                $useAutoCast = in_array($type, $importOptions->autoCastTypes(), true);
+                $isSameType = $type === $sourceColumn->getColumnDefinition()->getType();
+                if ($useAutoCast && !$isSameType) {
+                    if ($type === Snowflake::TYPE_OBJECT) {
+                        // object can't be casted from string but can be casted from variant
+                        $columnsSetSql[] = sprintf(
+                            'CAST(TO_VARIANT(%s) AS %s) AS %s',
+                            SnowflakeQuote::quoteSingleIdentifier($sourceColumn->getColumnName()),
+                            $destinationColumn->getColumnDefinition()->getSQLDefinition(),
+                            SnowflakeQuote::quoteSingleIdentifier($destinationColumn->getColumnName())
+                        );
+                        continue;
```

Comment:
ta metoda je takto nadesignovana, ale ten ty `continue` misto trochu lepe organizovaneho `if else` mi prijdou strasne neprehledne

Reply:
if else by byl hrozný hell, možeš to zkusit jak by to vypadalo :) Možná match(true) by to trochu pošéfoval, ale takto jdeš po těch podmínkách a máš early return přes continue. Když je to "flat" přes else if tak bude strašně komplikované ty elseif poskládat správně pod sebe.